ssaHAL/ssa/_action_handler.py

```
@@ -0,0 +1,251 @@
+        @param payload: The payload of the action.
+        """
+
+        if action_uri is None or len(action_uri) == 0:
+            print("[WARNING] Received message from invalid topic. Ignoring.")
+            return
+
+        # Check if the uri matches a top-level action
+        if action_uri in self.actions:
+            handler = self.actions[action_uri].callback
+            try:
+                handler(self._ssa, msg) # Invoke the action handler
```

Ruff (0.8.2)

Undefined name `msg`

(F821)